packages/relayer-lib/src/tx_builder/eth_to_cosmos.rs

```diff
@@ -256,95 +296,173 @@ where
+                tracing::info!(
+                    "Skipping update for slot {}",
+                    update.finalized_header.beacon.slot
+                );
+                continue;
+            }
 
-            tracing::debug!("Getting proof for block {}", block_hex);
-            let proof = self
-                .eth_client
-                .get_proof(&ibc_contract_address, vec![], block_hex)
-                .await?;
+            let update_next_sync_committee_agg_pubkey = update
+                .next_sync_committee
+                .clone()
+                .map(|sc| sc.aggregate_pubkey);
 
-            let account_update = AccountUpdate {
-                account_proof: AccountProof {
-                    proof: proof.account_proof,
-                    storage_root: proof.storage_hash,
-                },
-            };
+            // They are both options, so we can just compare them directly.
+            if update_next_sync_committee_agg_pubkey == current_next_sync_committee_agg_pubkey {
+                tracing::warn!(
+                    "Skipping header with same aggregate pubkey for slow {}",
+                    update.finalized_header.beacon.slot
+                );
+                continue;
+            }
 
-            let mut previous_period = 0;
-            let current_period = compute_sync_committee_period_at_slot(
+            // TODO: Not sure
+            let update_period = compute_sync_committee_period_at_slot(
                 ethereum_client_state.slots_per_epoch,
                 ethereum_client_state.epochs_per_sync_committee_period,
-                update.attested_header.beacon.slot,
+                update.finalized_header.beacon.slot,
             );
-            if current_period > 1 {
-                previous_period = current_period - 1;
-            }
-
-            tracing::debug!("Getting updates for previous period: {}", previous_period);
-
-            let previous_light_client_updates = self
-                .beacon_api_client
-                .light_client_updates(previous_period, 1)
-                .await?
-                .into_iter()
-                .map(|resp| resp.data)
-                .collect::<Vec<_>>();
-            let previous_light_client_update = previous_light_client_updates.first().unwrap();
-            let previous_next_sync_committee = previous_light_client_update
-                .next_sync_committee
-                .clone()
-                .unwrap();
-            if previous_next_sync_committee.aggregate_pubkey == prev_pub_agg_key {
-                tracing::debug!("Skipping header with same aggregate pubkey");
+            if update_period == latest_period {
+                tracing::info!(
+                    "Skipping header with same period for slot {}",
+                    update.finalized_header.beacon.slot
+                );
                 continue;
             }
+            latest_period = update_period;
```

Comment:
Shouldn't this be all the way at the end of the loop with the other assignments?